src/sentry/cache/redis.py

```diff
@@ -26,6 +27,8 @@ def __init__(self, **options):
     def set(self, key, value, timeout, version=None):
         key = self.make_key(key, version=version)
         v = json.dumps(value)
+        if len(v) > self.max_size:
+            raise ValueError('Cache key too large: %r %r' % (key, len(v)))
```

Comment:
I'm not entirely sure of the ramification of raising a exception here vs dropping it silently, then logging.

I'm a bit afraid about raising, then the task getting retried.